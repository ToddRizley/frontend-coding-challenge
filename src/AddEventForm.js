import React, { Component } from 'react';

class AddEventForm extends Component {

  render() {
    return (
      <div className='event-form row'>
        <h4>Add New Event</h4>
        <form className='form-group row col-xs-12' ref='eventForm' onSubmit={this.props.handleSubmit.bind(this)}>
          <div className='col-xs-3'>
            Event Name: <input name='event-title' className="form-control" placeholder='Add Event Title' type='text' required/>&nbsp;
          </div>
          <div className='col-xs-3'>
            Start: <input name='event-start' className="form-control" placeholder='Start Date/Time' type='datetime-local' required/>&nbsp;
          </div>
          <div className='col-xs-3'>
            End: <input name='event-end' className="form-control" placeholder='End Date/Time' type='datetime-local' required/>&nbsp;&nbsp;
          </div>
          <div className='col-xs-3'>
            <input id='add-event-button' type='submit' value='Submit'/>
          </div>
        </form>
      </div>
    );
  }
}

export default AddEventForm;
