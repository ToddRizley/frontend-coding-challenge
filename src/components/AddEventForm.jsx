import React from 'react';

const AddEventForm = props => (
  <div className="event-form container footer navbar-fixed-bottom">
    <h4>Add New Event</h4>
    <form
      className="form-group col-xs-12"
      onSubmit={props.handleSubmit.bind(this)}
    >
      <div className="col-xs-3">
            Title
            <input
              name="event-title"
              className="form-control"
              placeholder="Add Event Title"
              type="text"
              required
            />&nbsp;
      </div>
      <div className="col-xs-3">
            Start
            <input
              name="event-start"
              className="form-control"
              placeholder="Start Date/Time"
              type="datetime-local"
              required
            />&nbsp;
      </div>
      <div className="col-xs-3">
            End
            <input
              name="event-end"
              className="form-control"
              placeholder="End Date/Time"
              type="datetime-local"
              required
            />&nbsp;&nbsp;
      </div>
      <div className="col-xs-3">
        <input id="add-event-button" type="submit" value="Submit" />
      </div>
    </form>
  </div>
);


AddEventForm.propTypes = {
  handleSubmit: React.PropTypes.func,
};

export default AddEventForm;
