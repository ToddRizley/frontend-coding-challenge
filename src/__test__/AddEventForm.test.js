import React from 'react';
import AddEventForm from '../AddEventForm';
import renderer from 'react-test-renderer';

describe('<AddEventForm/>', () => {
  var handleSubmit = jest.fn()
   it('renders correctly', () => {
    const rendered = renderer.create(
      <AddEventForm handleSubmit={handleSubmit}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
