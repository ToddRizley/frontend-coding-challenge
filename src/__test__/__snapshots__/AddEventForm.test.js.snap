// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddEventForm/> renders correctly 1`] = `
<div
  className="event-form container footer navbar-fixed-bottom"
>
  <h4>
    Add New Event
  </h4>
  <form
    className="form-group col-xs-12"
    onSubmit={[Function]}
  >
    <div
      className="col-xs-3"
    >
      Title
      <input
        className="form-control"
        name="event-title"
        placeholder="Add Event Title"
        required={true}
        type="text"
      />
       
    </div>
    <div
      className="col-xs-3"
    >
      Start
      <input
        className="form-control"
        name="event-start"
        placeholder="Start Date/Time"
        required={true}
        type="datetime-local"
      />
       
    </div>
    <div
      className="col-xs-3"
    >
      End
      <input
        className="form-control"
        name="event-end"
        placeholder="End Date/Time"
        required={true}
        type="datetime-local"
      />
        
    </div>
    <div
      className="col-xs-3"
    >
      <input
        id="add-event-button"
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
